crates/revive-strategy/src/cheatcodes/mod.rs

```diff
@@ -1188,3 +1224,44 @@ fn get_context_ref_mut(
 ) -> &mut PvmCheatcodeInspectorStrategyContext {
     ctx.as_any_mut().downcast_mut().expect("expected PvmCheatcodeInspectorStrategyContext")
 }
+
+/// Applies REVM storage diffs to pallet-revive (REVM → pallet-revive sync)
+/// Note: Balance/nonce are NOT synced here as they're handled by migration in select_revive()
```

Comment:
i'm a bit confused as to why not migrate Balance and Nonce anyways?

Reply:
We need to sync it in vm.pvm(true/false) anyway, since it may be modified using cheatcodes. If we want to drop syncing when switching VMs, we also need to sync cheatcode execution between VMs, what we do not do it currently.